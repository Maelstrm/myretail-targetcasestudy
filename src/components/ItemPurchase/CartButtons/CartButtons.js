import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';

const styles = theme => ({
    root: {
    },
});

class CartButtons extends Component {

    render() {
        const { classes } = this.props;
        return (
            <Grid container item xs={12} style={{display:'flex'}} spacing={8}>
            <Grid item xs={6}>
                <Button variant="contained" color='primary' fullWidth style={{padding: '2px'}}>PICK UP IN STORE</Button>
            </Grid>
            <Grid item xs={6}>
                <Button variant="contained" color='primary' fullWidth>ADD TO CART</Button>
            </Grid>
            </Grid>
        );
    }
}

CartButtons.propTypes = {
    classes: PropTypes.object.isRequired,
};
export default withStyles(styles)(CartButtons);
