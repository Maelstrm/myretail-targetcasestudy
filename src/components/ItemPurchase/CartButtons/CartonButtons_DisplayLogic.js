import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

const displayThis = 2;

const  CartonButtons_DisplayLogic = () => {
    switch (displayThis) {
        case 1:
            return (
                <Grid container item xs={12} item xs={12} style={{display:'flex'}} spacing={8}>
                    <Grid item xs={6}>
                        <Button variant="contained" color='primary' fullWidth>PICK UP IN STORE</Button>
                    </Grid>
                    <Grid item xs={6}>
                        <Button variant="contained" color='primary' fullWidth>ADD TO CART</Button>
                    </Grid>
                </Grid>
            )
        case 2:
            return (
                <Grid container item xs={12} style={{display:'flex'}} spacing={8}>
                    <Grid item xs={6}>
                        <Button variant="contained" disabled fullWidth>PICK UP IN STORE</Button>
                    </Grid>
                    <Grid item xs={6}>
                        <Button variant="contained" disabled fullWidth>ADD TO CART</Button>
                    </Grid>
                </Grid>
            )
        default:
            return (
                <Grid container item xs={12} style={{display:'flex'}} spacing={8}>
                    <Grid item xs={6}>
                        <Button variant="contained" disabled fullWidth>PICK UP IN STORE</Button>
                    </Grid>
                    <Grid item xs={6}>
                        <Button variant="contained" disabled fullWidth>ADD TO CART</Button>
                    </Grid>
                </Grid>
            )
    }
};

const whichToDisplay = (purchasingCode) => {
    if(purchasingCode === 0 || 2 ) {
        displayThis = 1;
        return 1;
    }
    else {
        displayThis = 2;
        return 2
    }
};

const funcs = {
    whichToDisplay();
    CartonButtons_DisplayLogic();

};

export default funcs;