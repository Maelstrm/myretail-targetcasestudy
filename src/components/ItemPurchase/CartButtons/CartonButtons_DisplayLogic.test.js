import CartButtons from './CartButtons'
import Adapter from 'enzyme-adapter-react-16';

import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('tests', function() {
    it('renders correctly with 0', () => {
        const tree = shallow(<CartButtons data={0} />);
        expect(tree).toMatchSnapshot();
      });

      it('renders correctly with 1', () => {
          const tree = shallow(<CartButtons data={1} />);
          expect(tree).toMatchSnapshot();
      });
})


// test('check if purchaseChannelCode is working', () => {
//     expect(displaylogic(0)).toBe(
//         <Grid container item xs={12} item xs={12} style={{display:'flex'}} spacing={8}>
//             <Grid item xs={6}>
//                 <Button variant="contained" color='primary' fullWidth>PICK UP IN STORE</Button>
//             </Grid>
//             <Grid item xs={6}>
//                 <Button variant="contained" color='primary' fullWidth>ADD TO CART</Button>
//             </Grid>
//         </Grid>
//     );
//   });