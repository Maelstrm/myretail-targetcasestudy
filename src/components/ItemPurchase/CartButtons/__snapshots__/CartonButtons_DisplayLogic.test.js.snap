// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests renders correctly with 0 1`] = `
<div>
  <WithStyles(Grid)
    container={true}
    item={true}
    spacing={8}
    style={
      Object {
        "display": "flex",
      }
    }
    xs={12}
  >
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      <WithStyles(Button)
        color="primary"
        fullWidth={true}
        variant="contained"
      >
        PICK UP IN STORE
      </WithStyles(Button)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      <WithStyles(Button)
        color="primary"
        fullWidth={true}
        variant="contained"
      >
        ADD TO CART
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`tests renders correctly with 1 1`] = `
<div>
  <WithStyles(Grid)
    container={true}
    item={true}
    spacing={8}
    style={
      Object {
        "display": "flex",
      }
    }
    xs={12}
  >
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      <WithStyles(Button)
        disabled={true}
        fullWidth={true}
        variant="contained"
      >
        PICK UP IN STORE
      </WithStyles(Button)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      <WithStyles(Button)
        disabled={true}
        fullWidth={true}
        variant="contained"
      >
        ADD TO CART
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
